src/abstractsparsearrayinterface.jl

```diff
@@ -357,11 +357,13 @@ function sparse_mul!(
   β::Number=false;
   (mul!!)=(default_mul!!),
 )
+  storage(a_dest) .*= β
```

Comment:
```suggestion
  a_dest .*= β
```
Does this work? It should check that the broadcast call preserves the zero values and then just broadcast over the nonzero/stored values.

I'm ok using some "lower level" functionality like this if needed, but my goal was to try to make sure high level abstractions works in a smart way.

Also, an issue is that `storage` isn't part of the general API, and is really an implementation detail of `SparseArrayDOK`, so I think it should be `storedvalues(a_dest) .*= β`, if `a_dest .*= β` doesn't work for some reason.

Reply:
So `a_dest .*= beta` does not work (and actually just outputs wrong results in general). I'll open an issue:
```julia
julia> a
2×2 SparseMatrixDOK{Float64, typeof(SparseArraysBase.default_getunstoredindex)}:
 1.0  ⋅
  ⋅   ⋅

julia> a .*= 2
2×2 SparseMatrixDOK{Float64, typeof(SparseArraysBase.default_getunstoredindex)}:
 0.0  ⋅
  ⋅   ⋅
```

Reply:
Oh, yikes, that's a bad bug! Maybe something funny going on with the logic that is checking if the function preserves zero values, that part isn't so easy to get right...